### PR TITLE
DCOS-39066: Allow table to deal with dynamic data

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
To test please see the new storybook entry for Table > changing values

### CODE REVIEW

File: packages/table/stories/Table.stories.tsx [hunk 1]
```diff
@@ -63,10 +64,48 @@ const veryLongRenderer = () => (
...
+> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      items: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
```
Comment (weblancaster) [inline packages/table/stories/Table.stories.tsx]: Can we just have a `setInterval` with random data instead?

File: packages/table/stories/Table.stories.tsx [hunk 2]
```diff
@@ -63,10 +64,48 @@ const veryLongRenderer = () => (
...
 const empty = () => <Cell>empty</Cell>;
 const width = ({ width: totalWidth }: { width?: number }) =>
   totalWidth ? totalWidth * 0.3 : 100;
 
+const itemCellRenderer = item => (
+  <Cell>
+    <span>{item}</span>
+  </Cell>
+);
+class ChangingTable extends React.Component<
```
Comment (weblancaster) [inline packages/table/stories/Table.stories.tsx]: I suggest using a functional component that where it's props change randomly based on `setInterval`

**Comment by DanielMSchmidt** [comment]
I liked the idea and implemented it, please take another look @weblancaster

**Comment by DanielMSchmidt** [comment]
@Fabs I added a test case, please take another look :)

**Comment by juliangieseke** [comment]
whoever is cutting a release: please verify that `dist/packages/table/components/Cell.d.ts` imports are correct (no weit local path including your whole path) - looks like we have an issue on macOS :( (`npm run dist` on linux works)

**Pull request merged**